### PR TITLE
adds api for deleting an origin member

### PR DESCRIPTION
**Pull request by mwrock**
Signed-off-by: mwrock <matt@mattwrock.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by mwrock** [comment]
related to #2943

File: components/builder-protocol/protocols/originsrv.proto [hunk 1]
```diff
@@ -177,7 +186,7 @@ message OriginMemberListResponse {
 
 message OriginMemberRemove {
   optional uint64 origin_id = 1;
-  optional uint64 user_id = 2;
+  optional string account_name = 2;
```
Comment (raskchanky) [inline components/builder-protocol/protocols/originsrv.proto]: Am I remembering this wrong or were we not supposed to change existing protocol fields?

Comment (mwrock) [reply components/builder-protocol/protocols/originsrv.proto]: oh i should have mentioned. Nothing is using this message so this should be non breaking.

File: components/builder-depot/src/server.rs [hunk 2]
```diff
@@ -168,6 +168,20 @@ where
...
 }
 
+pub fn check_origin_owner<T>(req: &mut Request, account_id: u64, origin: T) -> IronResult<bool>
+where
+    T: ToString,
+{
+    match helpers::check_origin_owner(req, account_id, origin) {
+        Ok(b) => Ok(b),
+        Err(err) => {
+            let body = serde_json::to_string(&err).unwrap();
```
Comment (christophermaier) [inline components/builder-depot/src/server.rs]: Is `unwrap()` guaranteed to be safe here?

File: components/builder-originsrv/src/data_store.rs [hunk 3]
```diff
@@ -1465,6 +1465,16 @@ impl DataStore {
         Ok(())
     }
 
+    pub fn delete_origin_member(&self, omr: &originsrv::OriginMemberRemove) -> SrvResult<()> {
+        let conn = self.pool.get(omr)?;
+
+        conn.execute(
+            "SELECT * FROM delete_origin_member_v1($1, $2)",
```
Comment (christophermaier) [inline components/builder-originsrv/src/data_store.rs]: Since this doesn't return anything, you can drop the `* FROM` bit.

**Comment by reset** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/282534636).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/282534636).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**